schunk_libm5api/src/Device/Device.cpp

```diff
@@ -4805,21 +4806,26 @@ CDevice* newDevice(const char* acInitString)
                 printf("CDevice* newDevice(const char* acInitString): wrong format, no ':' found!\n");
 		return NULL;
         }
-
 	if( strcmp( pcToken, "RS232" ) == 0 )
 	{
 		return new CRS232Device();
 	}
+#ifdef USE_ESD
+	if( strcmp( pcToken, "ESD" ) == 0 )
+	{
+		return new CESDDevice();
+	}
+#endif
 #ifdef USE_PCAN
 	if( strcmp( pcToken, "PCAN" ) == 0 )
 	{
 		return new CPCanDevice();
 	}
 #endif
-#ifdef USE_ESD
-	if( strcmp( pcToken, "ESD" ) == 0 )
+#ifdef USE_SCAN

```

Comment:
I first tried to use USE_SOCKET_CAN but it was to long and it didn't work. This is why I shortened it and wrote USE_SCAN.